### ISSUE TITLE
SC2086: False positive for "export" and "local" built-ins (POSIX Shell)

### ISSUE THREAD

**Issue by itaranto** [koalaman/shellcheck#2443]
#### For bugs
- Rule ID:  SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
export FOO=$BAR
```

This shouldn't be a problem, it should expand the variable in the same way that `FOO=$BAR` does, i.e. quotes are not required here.

```sh
#!/bin/sh

# shellcheck disable=SC3043

do_something() {
    local foo=${1}
    echo "$foo"
}
```

I *know* that in POSIX `local` is not defined, but since almost every POSIX shell supports it I make an exception.

#### Here's what shellcheck currently says:

`SC2086 (info): Double quote to prevent globbing and word splitting.`

#### Here's what I wanted or expected to see:

*Nothing*

**Comment by jwilk** [comment]
Double quotes _are_ need in some shells. From example, this happens in Ubuntu 20.04 LTS, which ships with dash 0.5.10.2:
```console
$ dash -c 'v="hello world"; export w=$v; echo "$w"'
hello

$ dash -c 'v="hello world"; export w="$v"; echo "$w"'
hello world
```

This was fixed only in dash 0.5.11:
https://git.kernel.org/pub/scm/utils/dash/dash.git/commit/?id=cbb71a836874d176

Relevant POSIX bug:
https://www.austingroupbugs.net/view.php?id=351

**Comment by jwilk** [comment]
> And what about the `local` statement?

`local` is affected too:
```console
$ dash -c 'v="hello world"; f() { local w=$v; echo "$w"; }; f'
hello

$ dash -c 'v="hello world"; f() { local w="$v"; echo "$w"; }; f'
hello world
```

**Comment by climbTheStairs** [comment]
This also applies to SC2046:

Input:
```sh
#!/bin/sh
export FOO=$(my_command)
```
Output:
```
In test.sh line 2:
export FOO=$(my_command)
           ^-----------^ SC2046 (warning): Quote this to prevent word splitting.

For more information:
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
```
